tests/test-resize.lua

```diff
@@ -0,0 +1,412 @@
+    elseif #client.get() > 0 then
+
+        client.get()[1] : geometry {
+            x      = 200,
+            y      = 200,
+            width  = 300,
+            height = 300,
+        }
+
+        return true
+    end
+end)
+
+table.insert(steps, function()
+    root.fake_input("button_press",1)
+    local c = client.get()[1]
+
+    -- Just in case there is an accidental delayed geometry callback
+    assert(c:geometry().x      == 200)
+    assert(c:geometry().y      == 200)
+    assert(c:geometry().width  == 300)
+    assert(c:geometry().height == 300)
+
+    mouse.coords {x = 500+2*c.border_width, y= 500+2*c.border_width}
+
+    local corner = amouse.client.resize(c)
+
+    assert(corner == "bottom_right")
+
+    return true
+end)
+
+-- The geometry should remain the same, as the cursor is placed at the end of
+-- the geometry.
+table.insert(steps, function()
+
+    local c = client.get()[1]
+
+    assert(c:geometry().x      == 200)
+    assert(c:geometry().y      == 200)
+    assert(c:geometry().width  == 300)
+    assert(c:geometry().height == 300)
+
+    mouse.coords {x = 600+2*c.border_width, y= 600+2*c.border_width}
+
+    return true
+end)
+
+-- Grow the client by 100px
+table.insert(steps, function()
+
+    local c = client.get()[1]
+
+    assert(c:geometry().x      == 200)
+    assert(c:geometry().y      == 200)
+    assert(c:geometry().width  == 400)
+    assert(c:geometry().height == 400)
+
+    mouse.coords {x = 400+2*c.border_width, y= 400+2*c.border_width}
+
+    return true
+end)
+
+-- Shirnk the client by 200px
+table.insert(steps, function()
+
+    local c = client.get()[1]
+
+    assert(c:geometry().x      == 200)
+    assert(c:geometry().y      == 200)
+    assert(c:geometry().width  == 200)
+    assert(c:geometry().height == 200)
+
+    mouse.coords {x = 100, y= 100}
+
+    return true
+end)
+
+-- Grow the client by 100px from the top left
+table.insert(steps, function()
+
+    local c = client.get()[1]
+
+    assert(c:geometry().x      == 100)
+    assert(c:geometry().y      == 100)
+    assert(c:geometry().width  == 300)
+    assert(c:geometry().height == 300)
+
+    mouse.coords {x = 300, y= 200}
+
+    return true
+end)
+
+-- Shirnk the client by 100px from the top right
+table.insert(steps, function()
+
+    local c = client.get()[1]
+
+--             print(c:geometry().width)
+    assert(c:geometry().x == 100)
+    assert(c:geometry().y == 200)
+--     assert(c:geometry().width == 200-2*c.border_width) --FIXME off by border width...
+--     assert(c:geometry().height == 200-2*c.border_width) --FIXME off by border width...
+
+    mouse.coords {x = 300, y= 200}
+
+    return true
+end)
+
+--TODO test bottom left
+
+-- Stop the resize
+table.insert(steps, function()
+    root.fake_input("button_release",1)
+    root.fake_input("button_release",1)
+    root.fake_input("button_release",1)
+
+--     if not mousegrabber.isrunning then --FIXME it should work, but doesn't
+--         return true
+--     end
```

Comment:
There is something wrong. The resize should stop when the button is released, but it doesn't. I forgot about that FIXME, but it may be that the mousgrabber callback is called only when the mouse move, not when there is press/release (untested theory)